File: node-typescript/src/Transaction.ts
import Big from "big.js";
import { LocalDateTime } from "@js-joda/core";
import { TransactionType } from "./TransactionType";

//AKNBVHMN, 20/08/2020 13:14:11, 10.95, Kwik-E-Mart, REVERSAL, YGXKOEIA
export interface Transaction {
    readonly id: string;
    readonly transactedAt: LocalDateTime;
    readonly amout: Big;
    readonly merchantName: string;
    readonly type: TransactionType;
    readonly relatedTransactionId?: string;
}

File: transaction-analyzer/java/src/main/java/com/example/demo/Transaction.java
package com.example.demo;

import com.example.demo.TransactionType;

import java.math.BigDecimal;
import java.time.LocalDateTime;

//AKNBVHMN, 20/08/2020 13:14:11, 10.95, Kwik-E-Mart, REVERSAL, YGXKOEIA
public record Transaction(
        String id,
        LocalDateTime transactedAt,
        BigDecimal amount,
        String merchantName,
        TransactionType type,
        String relatedTransactionId
) {
}
